src/main/java/com/powsybl/openloadflow/ac/equations/AcEquationSystemUpdater.java

```diff
@@ -84,22 +44,48 @@ public void onShuntVoltageControlChange(LfShunt controllerShunt, boolean newVolt
         AcEquationSystem.updateShuntVoltageControlEquations(controllerShunt.getVoltageControl().orElseThrow(), equationSystem);
     }
 
+    private void updateElementEquations(LfElement element, boolean enable) {
+        // update all equations related to the element
+        for (var equation : equationSystem.getEquations(element.getType(), element.getNum())) {
+            if (equation.isActive() != enable) {
+                equation.setActive(enable);
+            }
+        }
+
+        // update all equation terms related to the element
+        for (var equationTerm : equationSystem.getEquationTerms(element.getType(), element.getNum())) {
+            if (equationTerm.isActive() != enable) {
+                equationTerm.setActive(enable);
+            }
+        }
+    }
+
     @Override
     public void onDisableChange(LfElement element, boolean disabled) {
+        updateElementEquations(element, !disabled);
         switch (element.getType()) {
             case BUS:
                 LfBus bus = (LfBus) element;
                 if (disabled && bus.isSlack()) {
                     throw new PowsyblException("Slack bus '" + bus.getId() + "' disabling is not supported");
                 }
-                bus.getVoltageControl().ifPresent(this::updateVoltageControl);
+                equationSystem.getEquation(bus.getNum(), AcEquationType.BUS_TARGET_PHI)
+                        .ifPresent(eq -> eq.setActive(!bus.isDisabled()));
+                equationSystem.getEquation(bus.getNum(), AcEquationType.BUS_TARGET_P)
+                        .ifPresent(eq -> eq.setActive(!bus.isDisabled() && !bus.isSlack()));
+                // set voltage target equation inactive, various voltage control will set next to the correct value
+                equationSystem.getEquation(bus.getNum(), AcEquationType.BUS_TARGET_V)
+                        .orElseThrow()
+                        .setActive(false);
+                bus.getVoltageControl().ifPresent(voltageControl -> AcEquationSystem.updateGeneratorVoltageControl(voltageControl, equationSystem));
```

Comment:
Do you miss `controllerBus.getReactivePowerControl().ifPresent` here?

Reply:
Yes, fixed